src/PowerShell/ErrorRecordFormatter.cs

```diff
@@ -0,0 +1,106 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+namespace Microsoft.Azure.Functions.PowerShellWorker.PowerShell
+{
+    using System;
+    using System.Linq;
+    using System.Management.Automation;
+    using System.Text;
+
+    internal class ErrorRecordFormatter
+    {
+        private const string TruncationPostfix = "...";
+        private const string Indent = "   ";
+
+        private readonly PowerShell _pwsh = PowerShell.Create();
+
+        public string Format(ErrorRecord errorRecord, int maxSize = 1 * 1024 * 1024)
```

Comment:
1024 here means just "K", as in "Kilobyte". Introducing a constant for it would add more harm than value: now you cannot confidently read the `OneKilo * OneKilo` value without looking up `OneKilo`... And what if someone changes this constant value...

The max size default here is somewhat arbitrary. 1 million characters are unlikely to cause troubles for the communication channels, and are in most cases enough to contain the formatted error (or at least the most important part of it). So, I chose the closest "round" number to 1 million, which happens to be 1024 * 1024, of course. :-)

The 1024 value below (_initial_ StringBuilder capacity) is unrelated. This is also a somewhat arbitrary number, not related to the first value. The only thing in common is that it also uses a conventional "round" number.

Reply:
Actually, now I want to add a comment on the maxSize purpose, as I realize the reasoning behind it is not entirely obvious.